lambda/custom/handlers.js

```diff
@@ -76,19 +80,57 @@ function newsInquiryIntentHandler() {
   return getTopHeadLine(topic)
     .then((topHeadlineArticle) => {
       topArticle = topHeadlineArticle;
-      return getArticleSummary(topArticle)
-        .then((topArticleSummary) => {
-          functionContext.response.speak(`the top headline is ${topArticle.title}<break time="1s"/> Here is the summary:<break time="1s"/>${topArticleSummary}`);
-          return functionContext.emit(':responseReady');
-        });
+      return getArticleSummary(topArticle);
+    })
+    .then((topArticleSummary) => {
+      functionContext.attributes.lastIntent = INTENT.NEWS_INQUIRY_INTENT;
+      functionContext.attributes.lastReadArticle = topArticle;
+      functionContext.response
+        .speak(`the top headline is ${topArticle.title}<break time="1s"/> Here is the summary:<break time="1s"/>${topArticleSummary}<break time="1s"/>would you like me to send the full article to your phone?`)
+        .listen('would you like to get the full article sent to your phone?');
+      return functionContext.emit(':responseReady');
     })
     .catch((err) => {
       logger.log('Error while looking for the summary of the article: ', err);
-      return this.emit(':tell', 'Sorry something is wrong on my side, please try again in a moment.');
+      this.response.speak('Sorry something is wrong on my side, please try again in a moment.');
```

Comment:
`functionContext.response.speak('...')`? since you aliased `this` to `functionContext`, might want to keep usage consistent. Unless the `this` in this scope is different.

Similarly on L70, L71, L73, L76.

Reply:
maybe instead you should remove `functionContext` in favour of `this` since all other functions are using `this`

Reply:
Good catch!!
Arrow function don't support this, so no need to use `functionContext` to protect this in the promises.
Code has been updated in the new commits (and tests are a more robust).